index.js

```diff
@@ -580,6 +582,12 @@ function Runtime() {
                                 cachedIvars = new ObjCIvars(self, classHandle());
                         }
                         return cachedIvars;
+                    case "$imageName":
+                        if (cachedImageName === null) {
+                            if (isClass())
+                                cachedImageName = api.class_getImageName(handle);
```

Comment:
Perhaps we could remove the `if` and use `classHandle()` instead of `handle`?

Reply:
will do, also found bug while writing the test :-)